client/consensus/slots/src/slots.rs

```diff
@@ -52,8 +53,6 @@ pub struct SlotInfo<B: BlockT> {
 	pub slot: Slot,
 	/// The instant at which the slot ends.
 	pub ends_at: Instant,
-	/// The inherent data.
-	pub inherent_data: InherentData,
```

Comment:
Should be replaced by `Box<InherentdataProvider>`

Reply:
Can you explain why? So far, it looks like it's not needed at all. In case I can fix the changes to make them generic without this field, of course.

Reply:
We should still create the inherent data providers inside here and just pass this around to create the inherent data. The function `create_inherent_data` should be made async. Then you can wait there for the provisioner result.

Reply:
But if we keep creating the inherent data providers here, does that not mean that we still do call the provisioner to do all the work for non block building validators? At least in polkadot as `polkadot_node_core_parachains_inherent::ParachainsInherentDataProvider::create`is called in `create_inherent_data_providers` https://github.com/paritytech/polkadot/blob/master/node/service/src/lib.rs#L1155-L1159. Which seems to be gathering the inherent data https://github.com/paritytech/polkadot/blob/master/node/core/parachains-inherent/src/lib.rs#L53.

I was thinking of factoring out the `slot`and `timestamp`calculation in to a different `async` in order to get these here and later on call `create_inherent_data_providers.

Reply:
Yeah, you will then just move the call to the provisioner to `create_inherent_data`. I don't see the problem.